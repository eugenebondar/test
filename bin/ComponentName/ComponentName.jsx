import React from 'react';
import classNames from 'classnames';
import './ComponentName.less'; //todo add there some styles or remove

export default class ComponentName extends React.PureComponent {

    static propTypes = {
        className: React.PropTypes.string,
        isError: React.PropTypes.bool,
        isLoading: React.PropTypes.bool,
        text: React.PropTypes.string
    };

    static defaultProps = {
        className : ''
    };

    render() {
        const finalClassName = classNames('component-name', this.props.className);
        return <div className={finalClassName}>
            Hello world
            {this.props.isError && <div>There was error</div>}
            {this.props.isLoading && <div>Loading</div>}
            <div>{this.props.text}</div>
        </div>;
    }
}
