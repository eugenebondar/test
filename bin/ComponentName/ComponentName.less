.component-name {
  //todo add here some styles or remove
}
