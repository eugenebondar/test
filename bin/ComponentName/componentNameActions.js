import Reflux from 'reflux';

const componentNameActions = {
    init: Reflux.createAction({ asyncResult: true }),
    destroy: Reflux.createAction({ sync: true })
};

const text = 'this is data coming from action';

componentNameActions.init.listen(function() {
    (Math.random() > 0.5) ? this.completed({ text }) : this.failed();
});

export default componentNameActions;
