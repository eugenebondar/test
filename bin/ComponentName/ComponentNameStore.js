import Reflux from 'reflux';
import componentNameActions from './componentNameActions';

export const defaultState = {
    isLoading: false,
    isError: false
};

class ComponentNameStore extends Reflux.Store {

    constructor() {
        super();

        this.listenables = componentNameActions;

        this.state = { ...defaultState };
    }

    /**
     * @listens componentNameActions.init
     */
    onInit() {
        this.setState({
            isLoading: true
        });
    }

    /**
     * @listens componentNameActions.init.completed
     */
    onInitCompleted({ text }) {
        this.setState({
            text,
            isLoading: false
        });
    }

    /**
     * @listens componentNameActions.init.failed
     */
    onInitFailed() {
        this.setState({
            isLoading: false,
            isError: true
        });
    }

    /**
     * @listens componentNameActions.destroy
     */
    onDestroy() {
        this.setState(defaultState);
    }
}

export default Reflux.initStore(ComponentNameStore);
