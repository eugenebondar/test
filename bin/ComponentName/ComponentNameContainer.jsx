import { Component } from 'reflux';
import React from 'react';
import ComponentNameStore from './ComponentNameStore';
import ComponentName from './ComponentName';
import componentNameActions from './componentNameActions';

export default class ComponentNameContainer extends Component.extend(React.PureComponent) {

    static propTypes = {};

    static defaultProps = {};

    constructor() {
        super();

        this.store = ComponentNameStore;

        this.mapStoreToState(ComponentNameStore, () => {
            //todo implement logic here or remove it
        });
    }

    componentDidMount() {
        componentNameActions.init();
    }

    componentWillUnmount() {
        super.componentWillUnmount();
        componentNameActions.destroy();
    }

    render() {
        return <ComponentName
            {...this.state}
        />;
    }
}
