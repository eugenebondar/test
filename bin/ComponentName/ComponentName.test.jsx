import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ComponentName from './ComponentName';

describe('ComponentName', () => {
    describe('with default set of props', () => {
        let component, errorConsole;

        beforeAll(() => {
            errorConsole = spyOn(console, 'error');
        });

        beforeEach(() => {
            component = shallow(<ComponentName/>);
        });

        it('matches snapshot', () => {
            expect(shallowToJson(component)).toMatchSnapshot();
        });

        it('prints no errors from propTypes validation', () => {
            expect(errorConsole).not.toHaveBeenCalled();
        })
    })
});
