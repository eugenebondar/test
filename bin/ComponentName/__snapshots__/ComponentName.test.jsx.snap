exports[`ComponentName with default set of props matches snapshot 1`] = `
<div
  className="component-name">
  Hello world
  <div />
</div>
`;
